### DISCUSSION TITLE
Editable Code Block RFC

### DISCUSSION THREAD

**Discussion by serifluous** [twilio-labs/paste#3341] [Review request]
Hi all,

Paste is adding an Editable Code Block component and we'd like your feedback. This will wrap the [Monaco Editor](https://microsoft.github.io/monaco-editor/) and provide basic functionality (error states, code folding, indentation guides). We're also creating a Paste code-editor library for you to create your own Paste-styled code editor if you need more features beyond what's provided through this component. 

You can think of this as a sibling component to our existing static [Code Block](https://paste.twilio.design/components/code-block).

![image](https://github.com/twilio-labs/paste/assets/3821840/fc9e52f7-9f01-47e0-a2c7-31d6420f506f)

## Component spec

Review [the component spec](https://docs.google.com/document/d/1VnycDxXHv6m1dII_iM3Xk3yDGKKNFZgXPkkenlpJlh4/edit?usp=sharing), and leave any comments/suggestions/questions directly in the google doc.

## Design

Review the below design proposal and leave any comments in this thread or in the [Figma file](https://www.figma.com/file/QDfUuqqCKijGeTrcISnjCh/Editable-Code-Block-explorations?type=design&node-id=71-66087&mode=design&t=INZoSOJxsUfXkucZ-4).

We're leaning toward keeping the visual style of the static Code Block and Editable Code Block fairly similar right now given scope and resource constraints. In the future, we can open up scope more and update Code Block and Editable Code Block together.

![image](https://github.com/twilio-labs/paste/assets/3821840/5db5a2f5-437f-457c-8a8a-8d2b3ca986f4)

Thanks!

**Comment by TheSisb** [comment]
As the component has now been released, the window for comment has now been closed.
